driver-core/src/main/com/mongodb/internal/connection/DefaultServerMonitor.java

```diff
@@ -244,11 +244,8 @@ private ServerDescription lookupServerDescription(final ServerDescription curren
                 }
             } catch (Throwable t) {
                 averageRoundTripTime.reset();
-                InternalConnection localConnection;
-                synchronized (this) {
```

Comment:
Any ideas why this was synchronized in the first place?  Looks like it was added to be overly defensive.

Reply:
The reply https://github.com/mongodb/mongo-java-driver/pull/1178#discussion_r1300315024 is equally applicable here.